pkg/balancer/consul_monitor.go

```diff
@@ -0,0 +1,80 @@
+package balancer
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/hashicorp/consul/api"
+	"github.com/square/p2/pkg/health"
+	"github.com/square/p2/pkg/logging"
+)
+
+type ConsulMonitor struct {
+	consul *api.Client
+	logger *logging.Logger
+}
+
+func NewConsulMonitor(cfg *api.Config, logger *logging.Logger) (Monitor, error) {
+	client, err := api.NewClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &ConsulMonitor{
+		client,
+		logger,
+	}, nil
+}
+
+func (c *ConsulMonitor) MonitorHosts(service string, strategy Strategy, quitCh <-chan struct{}) {
+	// strip everything after the first . in case the service passed is a vhost
+	service = strings.Split(service, ".")[0]
+
+	backoff := time.Duration(5)
```

Comment:
`backoff` is really a poll interval; you only start an exponential backoff when an error is returned. What do you hope to gain from an exponential backoff?

Since you refer to "5" and "30" (below) twice each, make them const values.